autorest/adal/token.go

```diff
@@ -554,11 +563,13 @@ func (spt *ServicePrincipalToken) InvokeRefreshCallbacks(token Token) error {
 }
 
 // Refresh obtains a fresh token for the Service Principal.
+// This method is not safe for concurrent use and should be syncrhonized.
 func (spt *ServicePrincipalToken) Refresh() error {
 	return spt.refreshInternal(spt.resource)
 }
 
 // RefreshExchange refreshes the token, but for a different resource.
+// This method is not safe for concurrent use and should be syncrhonized.
 func (spt *ServicePrincipalToken) RefreshExchange(resource string) error {
 	return spt.refreshInternal(resource)
 }
```

Comment:
Does it make sense to add synchronized methods here also ?

Reply:
I think for these "raw" methods the need for synchronization should be left up to the caller.  If they want a synchronized version they can call EnsureFresh().  I will update its comment to indicate it's safe for concurrent use.